core/network/src/packets.rs

```diff
@@ -2406,3 +2414,65 @@ pub struct EntityTeleport {
     pub pitch: u8,
     pub on_ground: bool,
 }
+
+#[derive(Default, AsAny, Clone)]
+pub struct Tags {
+    pub block_tags: Vec<(String, Vec<VarInt>)>,
+    pub item_tags: Vec<(String, Vec<VarInt>)>,
+    pub fluid_tags: Vec<(String, Vec<VarInt>)>,
+}
+
+impl Packet for Tags {
+    fn read_from(&mut self, buf: &mut Cursor<&[u8]>) -> anyhow::Result<()> {
+        for field in [
+            &mut self.block_tags,
+            &mut self.item_tags,
+            &mut self.fluid_tags,
+        ]
+        .iter_mut()
+        {
+            let length = buf.try_get_var_int()?;
+            for _tag in 0..length {
+                let identifier = buf.try_get_string()?;
+                let inner_length = buf.try_get_var_int()?;
+
+                let mut entries = Vec::with_capacity(inner_length as usize);
```

Comment:
Avoid calling `with_capacity` with untrusted inputs—it makes DoS attacks trivial :) A check here to ensure the length isn't too large would be good.

Reply:
Thanks, good tip! Though I'll just remove the `read_from` implementation completely, it's never needed anyway and the new packet lib is coming up

Reply:
Never needed as in this is actually a Server -> Client only packet